x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/allocation/SearchableSnapshotAllocator.java

```diff
@@ -341,11 +342,26 @@ private AllocateUnassignedDecision decideAllocation(RoutingAllocation allocation
                 );
                 return AllocateUnassignedDecision.yes(nodeWithHighestMatch.node(), null, nodeDecisions, true);
             }
+        } else if (isDelayedDueToShutdown(allocation, shardRouting)) {
+            return ReplicaShardAllocator.delayedDecision(shardRouting, allocation, logger, nodeDecisions);
         }
+
         // TODO: do we need handling of delayed allocation for leaving replicas here?
         return AllocateUnassignedDecision.NOT_TAKEN;
     }
 
+    private boolean isDelayedDueToShutdown(RoutingAllocation allocation, ShardRouting shardRouting) {
```

Comment:
```suggestion
    private boolean isDelayedDueToNodeRestart(RoutingAllocation allocation, ShardRouting shardRouting) {
```

Reply:
Forgot this in prior round, done now in df9115a0833a8344c9a1c38927b0029b10a2d896